components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,423 @@
+{
+    int bufferSize = httpd_req_get_url_query_len(HttpdRequest) + 1;     // +1 for null terminator
+    char queryBuffer[bufferSize];                                       // Buffer to store the query string
+    esp_err_t err = httpd_req_get_url_query_str(HttpdRequest, queryBuffer, sizeof(queryBuffer));
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "Failed to get the query string from the URL");
+        return ESP_FAIL;
+    }
+
+    bool isCodeFound = FindCode(queryBuffer, sizeof(queryBuffer));
+    if (!isCodeFound)
+    {
+        ESP_LOGE(TAG, "response does not include code at the beginning ");
+        return ESP_FAIL;
+    }
+
+    bool isCodeSent = xQueueSend(SendCodeFromHttpToTask, queryBuffer,  pdMS_TO_TICKS(SEC*15));
+    if (!isCodeSent)
+    {
+        ESP_LOGE(TAG, "Sent data with queue failed!");
+        return ESP_FAIL;
+    }
+
+    ESP_LOGI(TAG, "the CODE found in response");
+    return ESP_OK;
+}
+
+/**
+ * @brief Initialize and return an httpd_uri_t structure for the request URI.
+ * @param none
+ * @return An initialized httpd_uri_t structure with the specified URI, method, and handler.
+*/
+static httpd_uri_t RequestURIInit(void)
+{
+    const httpd_uri_t Request_Access_URI = {
+        .uri = AuthInterfaceHandler->ClientConfig.requestURI,
+        .method = HTTP_GET,
+        .handler = RequestDataAccess
+    };
+
+    return Request_Access_URI;
+}
+
+/**
+ * @brief Initialize and return an httpd_uri_t structure for the response URI.
+ * @param none
+ * @return An initialized httpd_uri_t structure with the specified URI, method, and handler.
+ */
+static httpd_uri_t ResponseURIInit(void)
+{
+    const httpd_uri_t Response_Access_URI = {
+        .uri = AuthInterfaceHandler->ClientConfig.responseURI,
+        .method = HTTP_GET,
+        .handler = HttpsCallbackHandler
+    };
+
+    return Response_Access_URI;
+}
+
+/**
+ * @brief This function check Time for token validation
+ * @return true if token expired, false otherwise
+ * */
+static bool Oauth_IsTokenExpired()
+{
+    bool tokenExpired = false;
+    uint32_t elapsedTime = (xTaskGetTickCount() - AuthPrivateHandler.TokenLastUpdate) 
+                            * portTICK_PERIOD_MS;             
+    elapsedTime = elapsedTime / 1000;                                           
+    if (elapsedTime > (HOUR - 300))
+    {
+        tokenExpired = true;  
+        ESP_LOGW(TAG , "Token is Expired");
+    }
+    return tokenExpired;
+}
+
+/**
+ * @brief This function extracts specific parameters from a JSON string and 
+ *         assigns them to corresponding fields in a TokenParam structure.
+ * @param[in] Json The input JSON string.
+ * @param[in] token address to save tokin extracted information.
+ * @return false if fail, true if finish successful.
+ */
+static bool ExtractTokenJson(char *Json, Token_t *token) 
+{
+    const char *TAG = "JsonExTraction";
+    cJSON *J_Token = cJSON_Parse(Json);
+
+    if (J_Token == NULL) 
```

Comment:
Please do all extraction in json.c file! I told you before

Reply:
It's a static function called just for ones so I prefer to keep it here in *.c file...It looks more optimal this way.